tests/test_sed.py

```diff
@@ -544,6 +544,56 @@ def test_SED_calculateSeeingMomentRatio():
 
     np.testing.assert_almost_equal(relative_size, num/den, 5)
 
+@timer
+def test_SED_sampleWavelength():
+    seed = 12345
+
+    sed  = galsim.SED(galsim.LookupTable([1,2,3,4,5], [0.,1.,0.5,1.,0.]),
+                      wave_type='nm', flux_type='fphotons')
+    bandpass = galsim.Bandpass(galsim.LookupTable([1,2,3,4,5], [0,0,1,1,0], interpolant='linear'), 
+                               'nm')
+    sedbp = sed*bandpass
+
+    out = sed.sampleWavelength(3,None)
+    np.testing.assert_equal(hasattr(sed,'deviate'),True,"Creating SED deviate cache failed.")
+    np.testing.assert_equal(len(sed.deviate),1,"Creating SED deviate failed.")
+
+    out = sed.sampleWavelength(3,None,seed=seed)
+    np.testing.assert_equal(len(sed.deviate),1,"Accessing existing SED deviate failed.")
+
+    test0 = np.array([ 4.15562438,  4.737775  ,  1.93594078])
+    np.testing.assert_array_almost_equal(out,test0,8,"Unexpected SED sample values.")
+
+    out = sed.sampleWavelength(3,bandpass,seed=seed)
+    np.testing.assert_equal(len(sed.deviate),2,"Creating new SED deviate failed.")
+
+    test1 = np.array([ 4.16227593,  4.6166918 ,  2.95075946])
+    np.testing.assert_array_almost_equal(out,test1,8,"Unexpected SED sample values.")
+
+    out = sed.sampleWavelength(1e3,bandpass,seed=seed,npoints=256)
+    np.testing.assert_equal(len(sed.deviate),2,"Unexpected number of SED deviates.")
+    np.testing.assert_equal(len(out),1e3,"Unexpected number of SED samples.")
+
+    np.testing.assert_equal(np.sum(out > sedbp.red_limit),0,
+                            "SED sample outside of function bounds.")
+    np.testing.assert_equal(np.sum(out < sedbp.blue_limit),0,
+                            "SED sample outside of function bounds.")
+
+    out2 = sed.sampleWavelength(1e3,bandpass,seed=seed,npoints=512)
+    np.testing.assert_equal(len(sed.deviate),3,"Unexpected number of SED deviates.")
+    np.testing.assert_almost_equal(out,out2,0,"SED sample npoints not equal.")
+
+    # Test the output distribution
+    out = sed.sampleWavelength(1e5,None,seed=seed)
+    x,step = np.linspace(sed.blue_limit,sed.red_limit,1e5,retstep=True)
+    c = (x[:-1]+x[1:])/2.
+    cdf = np.cumsum(sed(c)*step)
+    cdf /= cdf.max()
+
+    test = np.linspace(sed.blue_limit,sed.red_limit,100)
+    cdf1 = np.interp(test,c,cdf)
+    cdf2 = np.array([(out <= w).sum() for w in test],dtype=float)/len(out)
+    np.testing.assert_almost_equal(cdf1,cdf2,2,"Sampled SED does not match input.")
```

Comment:
You mentioned this yourself on the #822 issue page, but I wanted to second that it would be worth adding a test that uses a redshifted SED, since there is non-trivial code that depends on sed.redshift.